src/components/filters/FilterGroup/index.jsx

```diff
@@ -103,11 +103,23 @@ class FilterGroup extends React.Component {
 
   handleSectionClear(tabIndex, sectionIndex) {
     this.setState((prevState) => {
+      // update filter status
       const newFilterStatus = prevState.filterStatus.slice(0);
       newFilterStatus[tabIndex][sectionIndex] = {};
+
+      // update filter results; clear the results for this filter
+      let newFilterResults = prevState.filterResults;
```

Comment:
do we need to clone the object to be safe?
let newFilterResults =Object.assign({}, prevState.filterResults);

Reply:
Oh wow good catch! Yeah I think this is definitely a bug -- I think that it only works right now because filterStatus, which is cloned in L107, is also set in this state update, so the component always gets re-rendered despite the fact that filterResults hasn't 'changed' 😨